src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -1001,8 +1003,9 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VariantMetadata,
 
   def deleteVA(path: List[String]): (Type, Deleter) = vaSignature.delete(path)
 
-  def downsampleVariants(keep: Long): VariantSampleMatrix[T] = {
-    sampleVariants(keep.toDouble / countVariants())
+  def downsampleVariants(keep: Double): VariantSampleMatrix[T] = {
```

Comment:
Agree that fraction is better. I think it makes it clearer that you're not going to get exactly the number of variants you ask for.